schutzbot/scheduled_cloud_cleaner.sh

```diff
@@ -179,11 +179,36 @@ for snapshot in ${SNAPSHOTS}; do
 	START_TIME=$(echo "${snapshot}" | jq '.StartTime' | tr -d '"')
 
 	if [[ ${TAG} == "gitlab-ci-test" && ${TAG_VALUE} == "true" ]]; then
-		if [[ $(date -d "${START_TIME}" +%s) -lt $(date -d "- 2 minutes" +%s) ]]; then
+		if [[ $(date -d "${START_TIME}" +%s) -lt "${DELETE_TIME}" ]]; then
 			$AWS_CMD ec2 delete-snapshot --snapshot-id "${SNAPSHOT_ID}"
 			echo "The snapshot with id ${SNAPSHOT_ID} was deleted"
 		else
 			echo "The snapshot with id ${SNAPSHOT_ID} was created less than ${HOURS_BACK} hours ago"
 		fi
 	fi
 done
+
+# Remove tagged and old enough s3 objects
+AWS_BUCKET=composerredhat
+OBJECTS=$($AWS_CMD s3api list-objects --bucket ${AWS_BUCKET} | jq -c .Contents[])
+
+for object in ${OBJECTS}; do
+        LAST_MODIFIED=$(echo "${object}" | jq '.LastModified' | tr -d '"')
+        OBJECT_KEY=$(echo "${object}" | jq '.Key' | tr -d '"')
+
+        if [[ $(date -d "${LAST_MODIFIED}" +%s) -lt ${DELETE_TIME} ]]; then
+                TAG=$($AWS_CMD s3api get-object-tagging --bucket "${AWS_BUCKET}" --key "${OBJECT_KEY}" | jq .TagSet[0].Key | tr -d '"')
+                TAG_VALUE=$($AWS_CMD s3api get-object-tagging --bucket "${AWS_BUCKET}" --key "${OBJECT_KEY}" | jq .TagSet[0].Value | tr -d '"')
+
+                # divide in two ifs, so we dont perform so many requests
+                if [[ ${TAG} == "gitlab-ci-test" ]]; then
+                        if [[ ${TAG_VALUE} == "true" ]]; then
```

Comment:
I don't understand this separation. Why can't it  be a single if with `AND` for the conditions?